### DISCUSSION TITLE
Session Management in Android and iOS SDKs

### DISCUSSION THREAD

**Discussion by shahbaz17** [Web3Auth/Web3Auth#777] [Announcements]
Hello Web3Auth Mobile Devs 👋,

With our latest [Android SDK (v3.1.0)](https://github.com/Web3Auth/web3auth-android-sdk) and [Swift/iOS SDK (v5.0.0)](https://github.com/Web3Auth/web3auth-swift-sdk), we're introducing, ✨✨**Session Management**✨✨

The Session Management feature allows you to check the existing sessions with Web3Auth. The `web3Auth.sessionResponse()` in Android and `web3Auth?.state` will return the existing session of a user. To set the users’ session, pass `sessionTime` in the LoginParams, which will allow users to remain authenticated with Web3Auth for up to 1-day default, or a maximum of 7 days, or until they logout or session data is cleared.

Check out the documentation to try in your application.
Android: https://web3auth.io/docs/sdk/android
iOS: https://web3auth.io/docs/sdk/ios

**Comment by flodaniel** [comment]
Very cool!
Is there a roadmap when this is published in the flutter SDK?

**Reply by yashovardhan** [reply]
Expect that to be released by the end of this year.

**Comment by Syaba999** [comment]
Hi. Can we store the existing ed25519 private key with this SDK? We have users who have their own private key and we want to provide the ability to store the private key in the web3auth storage.
